node/tests/testcase.go

```diff
@@ -199,10 +199,7 @@ func (tcase testcase) Test(t *testing.T, np, npr datamodel.NodePrototype) {
 			t.Run("repr-create with AK+AV", func(t *testing.T) {
 				n3, err := shallowCopyMap(npr, n.(schema.TypedNode).Representation())
 				qt.Check(t, err, qt.IsNil)
-				// TODO: Improve checking mechanism; hits:
-				// -  expose datamodel.DeepEqual as a go-cmp Comparer, or
-				// - use the printer package and then do string diff on the results
-				qt.Check(t, datamodel.DeepEqual(n, n3), qt.IsTrue)
+				qt.Check(t, n3, NodeContentEquals, n)
```

Comment:
Note, there are other tests in `node` package that use `datamodel.DeepEqual` to assert equality of two nodes. I thought I check that reviewers are happy with this change before changing all assertions to use the new checker.

Reply:
SGTM, and this definitely leaves the tests in a better place long-term